.github/workflows/latest_deps.yml

```diff
@@ -0,0 +1,160 @@
+# People who are freshly `pip install`ing from PyPI will pull in the latest versions of all
+# dependencies. Since most CI runs are against the locked poetry environment, run
+# specifically against the latest dependencies to know if there's an upcoming breaking
+# change.
+#
+# To plug that hole, this job is a canary which periodically
+# - checks out develop,
+# - pip installs from source (building a wheel locally using `poetry-core`,
+#   and `pip install`ing from that), and
+# - runs test suites against that checkout.
```

Comment:
Sorry to still be confused here, are we actually testing against the *latest* dependencies or the *latest which meet any upper bounds we may have set*?

Reply:
No need to apologies. I'm lazily using the former wording to refer to the latter.

Reply:
```suggestion
# People who are freshly `pip install`ing from PyPI will pull in the latest versions of
# dependencies which match the broad requirements. Since most CI runs are against
# the locked poetry environment, run specifically against the latest dependencies to
# know if there's an upcoming breaking change.
#
# As an overview this workflow:
# - checks out develop,
# - pip installs from source (building a wheel locally using `poetry-core`,
#   and `pip install`ing from that), and
# - runs test suites against that checkout.
```

Do the steps in the workflow still match what is actually done?

Reply:
It's partially accurate.

- we're not using `poetry-core` to build wheels until #12478 lands.
- Otherwise I think it's accurate for `trial` and `systest`.
- mypy doesn't use a `pip` install, but the spirit is the same (install newest deps, rerun regular CI steps).

Let me see if I can tweak it a little...

Reply:
How does this read?

```suggestion
# People who are freshly `pip install`ing from PyPI will pull in the latest versions of
# dependencies which match the broad requirements. Since most CI runs are against
# the locked poetry environment, run specifically against the latest dependencies to
# know if there's an upcoming breaking change.
#
# As an overview this workflow:
# - checks out develop,
# - installs from source, pulling in the dependencies like a fresh `pip install` would, and 
# - runs mypy and test suites in that checkout.
```

Reply:
I think that sounds good. 👍